python/cudf/cudf/core/column_accessor.py

```diff
@@ -59,20 +118,49 @@ def copy(self):
         )
 
     def get_by_label(self, key):
-        result = self._grouped_data[key]
-        if isinstance(result, cudf.core.column.ColumnBase):
-            return self.__class__({key: result})
-        else:
-            result = _flatten(result)
-            if not isinstance(key, tuple):
-                key = (key,)
+        if isinstance(key, slice):
+            return self.get_by_label_slice(key)
+        elif isinstance(key, list):
```

Comment:
use a more general list-like class instead of just list, maybe `collections.abc.Sequence`?

Reply:
Use `pd.api.types.is_list_like`